### DISCUSSION TITLE
Questionable use of crypto

### DISCUSSION THREAD

**Discussion by ironhaven** [plausible/analytics#3556] [Self-Hosted Support]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

Plausible's main feature is that you can collect anonymized metrics by hashing PII with a salt that is rotated daily. In other words it should be impossible to create a hash collision between a previous stored hash and a new hash created with any user's PII. This allows a company to collect anonymized daily metrics securely.

The best way to hash data while adding a secret value is the HMAC algorithm. After reading the ingestion code it appears that instead of using HMAC from the Erlang crypto module, the code pulls a dependency in order to use SipHash as the keyed hash. It seems odd that SipHash was used as a message authentication code instead of as a hash table hashing function.

Unless there was a specific reason that SipHash was chosen to hash the data, HMAC_SHA256 should be used for message authentication purposes. I do not think that this SipHash construction is insecure, but without more documentation, this seems to be a suspicious choice.

A previous issue relating to crypto was posted [here](https://github.com/plausible/analytics/discussions/1152)

### Expected behavior

New documentation should be created to detail why cryptographic algorithms are chosen. Unless there is a good reason, HMAC-SHA256 should replace SipHash in data ingestion.

### Environment

```markdown
Any
```

**Comment by ruslandoga** [comment]
👋 @ironhaven 

Why would the hash need to be authenticated?

As for why SipHash is used instead of SHA256, it's probably due to size, the former fits in UInt64, the latter requires UInt256.
